Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -3326,13 +3325,13 @@ void RenderLayer::paintLayerContents(GraphicsContext& context, const LayerPainti
 
 #if ENABLE(LAYER_BASED_SVG_ENGINE)
         // SVG resource layers and their children are only painted indirectly, via paintSVGResourceLayer().
-        if (m_enclosingSVGResourceContainer) {
-            ASSERT(m_enclosingSVGResourceContainer->hasLayer());
-            return m_enclosingSVGResourceContainer->layer()->isPaintingSVGResourceLayer();
+        if (is<RenderSVGResourceContainer>(m_enclosingSVGHiddenOrResourceContainer)) {
+            ASSERT(m_enclosingSVGHiddenOrResourceContainer->hasLayer());
+            return m_enclosingSVGHiddenOrResourceContainer->layer()->isPaintingSVGResourceLayer();
         }
 
         // Hidden SVG containers (<defs> / <symbol> ...) and their children are never painted directly.
-        return !m_enclosingSVGHiddenContainer;
+        return !m_enclosingSVGHiddenOrResourceContainer;
```

Comment:
How about this:

```
if (!m_enclosingSVGHiddenOrResourceContainer)
    return true;

// Hidden SVG containers (<defs> / <symbol> ...) and their children are never painted directly.
if (!is<RenderSVGResourceContainer>(m_enclosingSVGHiddenOrResourceContainer))
    return false;

// SVG resource layers and their children are only painted indirectly, via paintSVGResourceLayer().
ASSERT(m_enclosingSVGHiddenOrResourceContainer->hasLayer());
return m_enclosingSVGHiddenOrResourceContainer->layer()->isPaintingSVGResourceLayer();
```
I think it is clearer if we split the case of `m_enclosingSVGHiddenOrResourceContainer` is null from the case it is not a resource container.